### DISCUSSION TITLE
How trully config conditions on Vue 3 / TS / Composition API?

### DISCUSSION THREAD

**Discussion by hrynevychroman** [stalniy/casl#799] [Q&A] [answered]
I want to add this library to my new project, but facing many problems with docs, don't know if it is not great documented or I don't see true example 🙂 

Need help with Conditions, added very simple example of library usage:

User json object: 
<img width="389" alt="image" src="https://github.com/stalniy/casl/assets/82209198/617fbc1f-d758-4a71-a27f-ce4768763a4b">

Update abilities on login and check if user can delete 'User', or user.value (where it is)
<img width="630" alt="image" src="https://github.com/stalniy/casl/assets/82209198/80cd4b7a-5846-4f15-9489-d3ae355eecfc">

For can('delete', 'User') -> i receive 'true', but i can delete only 'User' with firstname: 'Test', or not?
For can('delete', user.value) -> receive 'false', but this user having firstname === 'Test' (want to see true here and false on top example)

<img width="479" alt="image" src="https://github.com/stalniy/casl/assets/82209198/9dba09f8-2546-497d-9869-f64fd6e7f457">

Help me, please!
And Glory to Ukraine! 🇺🇦

**Comment by stalniy** [comment]
Hi Roman

The problem with good documentation is that it tends to be big. Nowadays people are overloaded by amount of information and obviously skip reading or miss important information. Anyway, it’s easier to ask :)
  
There is even a special term for this - help vampire. 

You can find information you need in this section - https://casl.js.org/v6/en/guide/intro#checking-logic

when you check an actual object, casl needs  to understand its type. If it’s just a POJO, there is no way to know it’s type in runtime. 

I’d recommend to carefully go through the guide and you will find all answers to your questions. 

Glory to heroes!

**Reply by hrynevychroman** [reply, marked as answer]
Yes, sorry, read docs one more time and everything works now, problem was in "Subject type detection" and how exactly I need to provide it to CASL. So if anyone will face this problem next time, everything you need to do -> wrap your user object inside `subject` function, to provide type declaration 👇 
<img width="827" alt="image" src="https://github.com/stalniy/casl/assets/82209198/a4bd7bc6-bb8e-4454-8108-cba09b022311">

Link to docs: https://casl.js.org/v6/en/guide/subject-type-detection

Some more words to you personally @stalniy : Дуже дякую за таку чудову бібліотеку, радий що такий продукт зроблений українцем для всього світу! Гарного Вам дня та всього найкращого!
